File: Server/src/main/java/com/api/server/foodb/CiteThisExternalLink.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "cite_this_external_links")
public class CiteThisExternalLink {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 1000)
    @NotNull
    @Column(name = "url", nullable = false, length = 1000)
    private String url;

    @Size(max = 255)
    @NotNull
    @Column(name = "name", nullable = false)
    private String name;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Size(max = 255)
    @Column(name = "ref_id")
    private String refId;

}

File: DotNetServer/DotNetServer/Entities/Foodb/CiteThisExternalLink.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNetServer.Entities.Foodb;

[Table("cite_this_external_links")]
public partial class CiteThisExternalLink
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("url")] [StringLength(1000)] public string Url { get; set; } = null!;

    [Column("name")] [StringLength(255)] public string Name { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("ref_id")] [StringLength(255)] public string? RefId { get; set; }

    [InverseProperty("ExternalLink")]
    public virtual ICollection<CiteThisExternalLinkReferencing> CiteThisExternalLinkReferencings { get; set; } =
        new List<CiteThisExternalLinkReferencing>();
}